.travis.yml

```diff
@@ -1,49 +1,51 @@
 language: android
-dist: trusty
-sudo: required
+# These two lines probably don't change anything
+dist: trusty # choose the type of operating system (Ubuntu 14)
+sudo: required # toggle sudo mode (should be a bit faster)
 android:
   components:
-    # The BuildTools version used by your project
     - tools
     - platform-tools
-    - tools
-    - build-tools-27.0.3
-    # The SDK version used to compile your project
-    - android-28
-    # The SDK version used by the system image
-    # - android-22
-    - android-19
-    - extra-google-google_play_services
-    - extra-google-m2repository
-    - extra-android-m2repository # for design library
-    - addon-google_apis-google-19 # google play services
-    - sys-img-armeabi-v7a-google_apis-19
+    - tools # needed twice because travis is stupid
+    - build-tools-27.0.3 # your build tools version
+    - android-28 # your app's target api level
+    - android-19 # your emulator's api level
+    - android-21
```

Comment:
Can you elaborate on why you need `tools` twice and three different Android SDKs? If it solves a specific problem, other teams might benefit from it.